### PR TITLE
CONTRIBUTING refresh for training-kit

### PR DESCRIPTION
**Pull request by crichID**
## TL;DR

Adding a more robust CONTRIBUTING guide per our aspirations in #415.

### Next Steps

I would like to study a few more CONTRIBUTING guides and consider additional topics to discuss here. Including:

- [atom/atom](https://github.com/atom/atom/blob/master/CONTRIBUTING.md)
- [opengovernment](https://github.com/opengovernment/opengovernment/blob/master/CONTRIBUTING.md)
- [codebuddies](https://github.com/codebuddiesdotorg/codebuddies/blob/master/contributing.md) - per @hollenberry comment in #440 
- [hoodie](https://github.com/hoodiehq/hoodie/blob/master/CONTRIBUTING.md) - per @hollenberry comment in #440
- In the CONTRIBUTING guide, explain your label system and include search links - [for example](https://github.com/github/training-kit/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)

I also want to compare it to these templates/checklists for additional inspiration:

- [ ] [opensource.guide](https://opensource.guide/starting-a-project/#writing-your-contributing-guidelines)
- [ ] @nayafia's [CONTRIBUTING-template](https://github.com/nayafia/contributing-template/blob/master/CONTRIBUTING-template.md)
- [ ] mozilla's guide for [How to Build a CONTRIBUTING.md](http://mozillascience.github.io/working-open-workshop/contributing/)

@github/services-training I welcome your :eyes:, ✏️ , and 💡 on what we want to add here.

### CODE REVIEW

**Comment by JasonEtco** [comment]
FWIW, I added a note about using Primer in #567 (see cabe2ff7121d0480c2d295b430d13b5ccffc0035)

**Comment by hectorsector** [comment]
@crichID I have incorporated the meat of https://github.com/github/training-kit/pull/436 into this PR. I think it makes more sense for those guidelines to be included here rather than as a standalone page on the site. What do you think?

**Comment by crichID** [comment]
> I think it makes more sense for those guidelines to be included here rather than as a standalone page on the site. What do you think?

@hectorsector that works for me :+1: thank you for adding these.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,9 +1,149 @@
...
+
+When you are ready to test your changes, you will want to build the repository locally. This is fully automated through a series of shell scripts based [the scripts to rule them all](https://github.com/github/scripts-to-rule-them-all)!
+
+To build of the materials do the following:
+
+1. Run `script/setup`
+1. Run `sass --watch assets/_scss/main.scss:assets/css/main.css` to compile the SCSS
+1. Run `script/server`
+    - When successful, the script will initiate a local server at `http://127.0.0.1:4000/on-demand`.
+    - The `/on-demand/` directory is a result of the [`baseurl` variable](https://byparker.com/blog/2014/clearing-up-confusion-around-baseurl/) provided by Jekyll. If you'd like to serve your site under a different directory, change it in [`_config.yml`](https://github.com/github/training-kit/blob/contrib-guide/_config.yml).
```
Comment (crichID) [inline CONTRIBUTING.md]: @hectorsector should we specify which types of files/links Jekyll will and will not touch?

Comment (hectorsector) [reply CONTRIBUTING.md]: I'm hesitant to go into the deep end here, @crichID, since it starts uncovering some Jekyll specific things. I do think, however, that it's a solid topic for an on-demand class on Jekyll and advanced GitHub Pages.

Comment (crichID) [reply CONTRIBUTING.md]: @hectorsector :+1: fair point! Thank you for making these additions here.

**Comment by hollenberry** [comment]
@crichID 👋 ! 

Do you think this is ready to go, or are there a few loose ends that still need adding?

**Comment by crichID** [comment]
> Do you think this is ready to go, or are there a few loose ends that still need adding?

@hollenberry I think this is a good start. Let's :ship: this and continue to iterate  in future PRs as needed.
